### PR TITLE
Use UUID v4 for generating operation IDs

### PR DESCRIPTION
**Pull request by TimGuiteDiamond**
I think this should fulfill the same requirement but will be more useful if many subscriptions are opened.

### CODE REVIEW

**Comment by ecthiender** [comment]
@TimGuiteDiamond according to the [apollo protocol](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) the operation id has to be unique only per connection. Do you think we gain any extra by using UUIDs?

**Comment by TimGuiteDiamond** [comment]
I was looking at this because I was interested in having *many* simultaneous subscriptions in a single thread, each of which would have it's own callback function, stored in a lookup table with the `id` as key (potentially a PR coming on this too 👍).

In this case, I wanted to be very sure that subscriptions wouldn't clash on `id` value, and as `UUID` is used in multiple domains and part of the standard library it seemed a good, stable and robust choice.

**Comment by ecthiender** [comment]
@TimGuiteDiamond okay. I don't feel particularly too strong about this. But somehow it feels like it might be overkill to generate UUIDs for this. But in any case, if you resolve the conflicts, I'll merge this PR.

**Comment by ecthiender** [comment]
> having many simultaneous subscriptions in a single thread, each of which would have it's own callback function, stored in a lookup table with the id as key 

This is already possible. Consider the following code:

```python
from graphql_client import GraphQLClient

ws = GraphQLClient('ws://localhost:8080/graphql')

def notification_handler(_id, data):
  print(f"Got new notification: {data}")

def show_online_users(_id, data):
  print(f"Online Users: {data.onlineUsers}")

latest_notifications = """
  subscription getNotifications {
    notifications(last: 10) {
      id
      message
    }
  }
"""

sub_id = ws.subscribe(latest_notifications, callback=notification_handler)

online_users = """
  subscription getOnlineUsers {
    onlineUsers {
      username
      lastSeen
    }
  }
"""

sub_id2 = ws.subscribe(onilne_users, callback=show_online_users)

# later stop the subscription
ws.stop_subscribe(sub_id)
ws.stop_subscribe(sub_id2)

ws.close()
```

File: graphql_client/__init__.py [hunk 1]
```diff
@@ -50,7 +51,7 @@ def _conn_init(self, headers=None):
         self._conn.recv()
 
     def _start(self, payload):
-        _id = gen_id()
+        _id = uuid.uuid4()
```
Comment (ecthiender) [inline graphql_client/__init__.py]: @TimGuiteDiamond have you tested this code? This throws a runtime exception, as the UUID object is not J SON-serializable. You need to fix this to:

```suggestion
        _id = uuid.uuid4().hex
```

**Review by ecthiender — CHANGES_REQUESTED** [review]

File: graphql_client/__init__.py (hunk 1 above)
Comment (TimGuiteDiamond) [reply graphql_client/__init__.py]: Sorry about this! I just made this change quickly and it looked like it returned a string, I'll make the change tomorrow.

I have some suggestions on how to do the multiple subscriptions but I'll leave them for a different PR/issue.

Thanks for responding and working on this 😄

Comment (TimGuiteDiamond) [reply graphql_client/__init__.py]: Have made this change now and tested it with my system.

Sorry for the extra commits, accidentally formatted it with black, but I reverted it so should just be the minimal changes now 👍

**Comment by ecthiender** [comment]
Hey @TimGuiteDiamond , will you resolve the conflicts as well, please? Then I can merge this.

**Comment by TimGuiteDiamond** [comment]
Don't merge that yet, made a mistake merging the latest master!

**Comment by TimGuiteDiamond** [comment]
I think this should do the job 👍

**Comment by ecthiender** [comment]
@TimGuiteDiamond thanks for all the work! Please do open an issue to discuss the multiple subscriptions thing you were talking about. Happy to discuss it on an issue thread.

**Review by ecthiender — APPROVED** [review]

**Comment by TimGuiteDiamond** [comment]
You're welcome! This was quite a small change really but I am using this package as part of a bigger project at the moment so happy to contribute 😄

**Pull request merged**